orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/ParallelDeployStage.groovy

```diff
@@ -61,12 +61,13 @@ class ParallelDeployStage implements StageDefinitionBuilder {
 
   @CompileDynamic
   protected Collection<Map<String, Object>> parallelContexts(Stage stage) {
+    def defaultStageContext = new HashMap(stage.context)
```

Comment:
previously there was a bunch of nastiness in here that was mutating the original stage (including changing the type so that a different afterStages call would accidentally get invoked when the stage completed which was the thing breaking locking)